services/collaboration/pkg/connector/connector.go

```diff
@@ -33,6 +38,61 @@ func NewResponseWithLock(status int, lockID string) *ConnectorResponse {
 	}
 }
 
+// NewResponseLockConflict creates a new ConnectorResponse with the status 409
+// and the "X-WOPI-Lock" header having the value in the lockID parameter.
+//
+// This is used for conflict responses where the current lock id needs
+// to be returned, although the `GetLock` method also uses this method for a
+// successful response (with the lock id included)
+// The lockFailureReason parameter will be included in the "X-WOPI-LockFailureReason".
+func NewResponseLockConflict(lockID string, lockFailureReason string) *ConnectorResponse {
+	return &ConnectorResponse{
+		Status: 409,
+		Headers: map[string]string{
+			HeaderWopiLock:              lockID,
+			HeaderWopiLockFailureReason: lockFailureReason,
+		},
+	}
+}
+
+// NewResponseWithVersion creates a new ConnectorResponse with the specified status
+// and the "X-WOPI-ItemVersion" header having the value in the mtime parameter.
+func NewResponseWithVersion(status int, mtime *types.Timestamp) *ConnectorResponse {
+	return &ConnectorResponse{
+		Status: status,
+		Headers: map[string]string{
+			HeaderWopiVersion: helpers.GetVersion(mtime),
+		},
+	}
+}
+
+// NewResponseConflictWithVersion creates a new ConnectorResponse with the status 409
+// and the "X-WOPI-ItemVersion" header having the value in the mtime parameter.
+// The lockFailureReason parameter will be included in the "X-WOPI-LockFailureReason".
+func NewResponseConflictWithVersion(mtime *types.Timestamp, lockFailureReason string) *ConnectorResponse {
```

Comment:
I'm pretty sure all the 409 responses should have the corresponding `X-WOPI-Lock` header... code at https://github.com/owncloud/ocis/pull/9686/files#diff-57a0aa20d4a5545fa33573bfb8120aea99fc483bc370054528f0e8e0629728f1R431 was sending an empty lock, but the header was sent.

This could be merged with the `NewResponseWithVersionAndLock` to have less methods around.

Reply:
We need no version here, i used the normal LockConflict response. `NewResponseWithVersionAndLock` is obsolete, deleting it.